packages/prop-types/prop-types.js

```diff
@@ -0,0 +1,49 @@
+/* eslint-disable import/prefer-default-export */
+
+export const componentWithName = passedName => {
+  if (typeof passedName !== 'string') {
+    throw new Error('passedName must be a string')
+  }
+  const checkProp = (props, propName, componentName) => {
+    if (Array.isArray(props[propName])) {
+      return props[propName]
+        .map((_, index) => checkProp(props[propName], index, componentName))
```

Comment:
Add a comment above this:

// Iterates through every child to see if one of the elements does not match the passed name